siriuspy/siriuspy/oscilloscope/keysight.py

```diff
@@ -0,0 +1,203 @@
+        sock = self._socket
+
+        # Get the number of waveform points
+        sock.sendall(b":WAVeform:POINts?\n")
+        points = int(sock.recv(1024).decode('ascii'))
+        print('Points:', points)
+
+        # Get sample rate
+        sock.sendall(b":ACQuire:SRATe?\n")
+        srate = float(sock.recv(1024).decode('ascii'))
+        print('Sample rate:', srate)
+
+        # Get bandwidth
+        sock.sendall(b":ACQuire:BANDwidth:FRAMe?\n")
+        bdw = float(sock.recv(1024).decode('ascii'))
+        print('Bandwidth:', bdw)
+
+        # Set the waveform channel source
+        sock.sendall(b":WAVeform:SOURce " + channel.encode('ascii')+b"\n")
+        sock.sendall(b":WAVeform:SOURce?\n")
+        channel = sock.recv(1024).decode('ascii')
+
+        # Get scales
+        sock.sendall(b":WAVeform:XINCrement?\n")
+        xinc = float(sock.recv(1024).decode('ascii'))
+        print('Horizontal Scale:', xinc)
+        sock.sendall(b":WAVeform:YINCrement?\n")
+        yinc = float(sock.recv(1024).decode('ascii'))
+        sock.sendall(b":WAVeform:YORigin?\n")
+        yor = float(sock.recv(1024).decode('ascii'))
+        print('Vertical Scale:', xinc, yor)
+
+        # Data aquisition
+        sock.sendall(b":WAVeform:STReaming OFF\n")
+        sock.sendall(b":WAVeform:DATA?\n")
+        _ = sock.recv(1)  # ignore marker '#'
+        num = int(sock.recv(1).decode('ascii'))
+        datanum = int(sock.recv(num).decode('ascii'))
+        dataraw = b''
+
+        # Return scope to RUN mode
+        self._socket.sendall(b":RUN\n")
+
+        while len(dataraw) < datanum:
+            dataraw = dataraw + sock.recv(datanum)
+        dataraw = dataraw[0:-1]  # remove EOF char
+
+        va1 = _np.array(list(dataraw)[0::2])
+        va0 = _np.array(list(dataraw)[1::2])
+        datay = ((va1 << 8) + va0 - 2**16*(va1 >> 7)) * yinc + yor
+
+        datax = _np.arange(datay.size)*xinc
+        return datax, datay, srate, bdw
+
+    def wfm_get_data(self, channel=None):
+        """."""
+        channel = channel or self.chan
+        self.connect()
+        wavet = None
+        waved = None
+        try:
+            self.wfm_init()
+            self.wfm_config()
+            tini = _time.time()
+            print('Acquiring ' + self.chan)
+            wavet, waved, srate1, bdw1 = self.wfm_acquire(channel)
+            print('Total acquisition time:', _time.time() - tini)
+            # self._socket.sendall(b":WAVeform:STReaming ON\n")
+        except Exception:
+            print("Unexpected error:", _sys.exc_info()[0])
+            print('Close connetion by exception')
+        finally:
+            self.close()
+
+        return wavet, waved
+
+    def stat_init(self):
+        """Enable scope measurement statistics info."""
+        sock = self._socket
+        # Set bit order to MSB First
+        sock.sendall(b":MEASure:STATistics ON\n")
+        sock.sendall(b":MEASure:SENDvalid ON\n")
+
+    def stat_acquire(self):
```

Comment:
at a first sight I thought that there was a typo here and it should be "start_acquire", then I saw the docstring explaining that `stat` is a short for "statistics". Maybe rename this methods to avoid confusion? `acquire_stats` or `stats_acquire`. Also the other method to: `init_stats`, `stats_init`, for this case I would suggest something like `enable_stats` following the docstring description.

Reply:
"we'll do"